### PR TITLE
Added graphviz to the conda environment file

### PR DESCRIPTION
**Pull request by ayshih**
Oops, left out `graphviz` in #6479.  `graphviz` is in fact one of the primary reasons that we'd want to utilize `conda`.  I also added two `pytest` extensions that are used by the test commands in our `tox` builds.

### CODE REVIEW

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Huh:

```
INTERNALERROR> OSError: An attempt was made to connect to the internet by a test that was not marked `remote_data`. The requested host was:
Internet access disabled
```

**Comment by ayshih** [comment]
This use of the network stack is set up at configuration, so it doesn't matter if the rerun option is not used.

File: tox.ini [hunk 1]
```diff
@@ -125,6 +125,8 @@ deps =
 conda_env = sunpy-dev-env.yml
 install_command = pip install --no-deps --no-build-isolation {opts} {packages}
 allowlist_externals = conda
+# Remove pytest-rerunfailures to avoid appearing to access the internet when pytest-xdist is used
```
Comment (nabobalis) [inline tox.ini:128]: Can we not move this into the command block or will tox error?

**Comment by ayshih** [comment]
I'm trying out the different approach of uninstalling `pytest-rerunfailures` prior to running the test c9mmand.

File: tox.ini (hunk 1 above)
Comment (ayshih) [reply tox.ini:128]: I don't know: I still don't run tox locally, and I don't see guidance in the tox documentation.  Maybe it'd work fine?

Comment (nabobalis) [reply tox.ini:128]: Want me to try?

**Comment by nabobalis** [comment]
Failure is due to `pytest-remotedata-0.3.2`, 0.3.3 fixed the deprecation warning but isn't on conda-forge?

**Comment by nabobalis** [comment]
Wait, https://github.com/conda-forge/pytest-remotedata-feedstock/blob/main/recipe/meta.yaml is up to date.

**Comment by ayshih** [comment]
Yeah, something about the removal is trigger a re-solve, and consequently a regression in `pytest-remotedata`...

**Comment by ayshih** [comment]
No, no, I can get this to work. =)

**Comment by nabobalis** [comment]
I can run any tox configs locally if you would like.

**Comment by ayshih** [comment]
I thought making sure the solver prioritized `conda-forge` was a good thing, but apparently not!  Let's try skipping the solver entirely...

**Comment by ayshih** [comment]
Yay, passing again

**Review by nabobalis — APPROVED** [review]

**Pull request merged**